### PR TITLE
migrate access_private to conan v2

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
This conan v2 migration PR was generated by automatoc script version 4.
It uses simple find and replace technics. Feel free to extend it.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/access_private/all/test_package/conanfile.py [hunk 1]
```diff
@@ -13,6 +14,6 @@ def build(self):
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self.settings):
+        if not tools.build.cross_building(self, self.settings):
```
Comment (ericLemanissier) [inline recipes/access_private/all/test_package/conanfile.py:17]:
```suggestion
        if not tools.build.cross_building(self):
```
This is still incorrect

Comment (AndreyMlashkin) [reply recipes/access_private/all/test_package/conanfile.py:17]: Okey, weired. Anyway, it is fixed in a later script version :)

**Comment by ericLemanissier** [comment]
> This conan v2 migration PR was generated by automatoc script version 4.
> It uses simple find and replace technics. Feel free to extend it.

where can it be found ?

**Comment by AndreyMlashkin** [comment]
> > This conan v2 migration PR was generated by automatoc script version 4.
> > It uses simple find and replace technics. Feel free to extend it.
> 
> where can it be found ?

It's not checked in yet. It's just a local bash script on my machine. I can share it, if you want

**Comment by AndreyMlashkin** [comment]
https://github.com/AndreyMlashkin/conan-center-index/tree/conan_v2_migration_backup_try5

This script compiles a recipe locally:
https://github.com/AndreyMlashkin/conan-center-index/blob/conan_v2_migration_backup_try5/recipes/create_recipes.sh

This script creates branches for successfull builds:
https://github.com/AndreyMlashkin/conan-center-index/blob/conan_v2_migration_backup_try5/create_branches.sh

It's a very raw draft and wasn't mean't to share with anyone

**Comment by AndreyMlashkin** [comment]
Find and replace templates are taken from a table here:
https://github.com/conan-io/conan-center-index/blob/master/docs/v2_linter.md

**Comment by ericLemanissier** [comment]
This should be closed in favor of #12786 which is a more complete modernization

**Pull request closed without merging**